### PR TITLE
fix(markdoc): remove horizontal rules

### PR DESCRIPTION
**Pull request by noon-dawg**
This pull request removes horizontal rules from the readability logic.

### CODE REVIEW

File: src/readability.ts [hunk 1]
```diff
@@ -374,6 +386,7 @@ export function preprocessMarkdown(markdown: string) {
         .use(addPeriodsToListItems)
         .use(removeJsItems)
         .use(removeUnwantedNodeTypes)
+        .use(removeHorizontalRules)
```
Comment (Weetbix) [inline src/readability.ts:389]: @noon-dawg Ah wait sorry, you'll see this breaks the existing YAML front matter unit test. 

Likely you can swap the order so `removeHorizontalRules` plugin is used after `removeFrontmatter`. 

Let me know if that doesn't work and we can think of another approach :)

Comment (noon-dawg) [reply src/readability.ts]: Thank works!! Thank you 🙏

**Comment by noon-dawg** [comment]
(Added a test in [928fb1d](https://github.com/Rebilly/lexi/pull/227/commits/928fb1df5ae91b535560f17b03c429c741dd5807))

File: src/readability.test.ts [hunk 2]
```diff
@@ -255,6 +255,28 @@ This is the only content.
         `);
     });
 
+    it('should horizontal rules', () => {
```
Comment (Weetbix) [inline src/readability.test.ts:258]:
```suggestion
    it('should remove horizontal rules', () => {
```

File: src/readability.test.ts [hunk 3]
```diff
@@ -255,6 +255,28 @@ This is the only content.
...
+        const stripped = preprocessMarkdown(
+            `
+            ---
```
Comment (Weetbix) [inline src/readability.test.ts:261]:
```suggestion
---
```

**Review by Weetbix — CHANGES_REQUESTED** [review]
Looking good!

File: src/readability.test.ts (hunk 2 above)
Comment (noon-dawg) [reply src/readability.test.ts:258]: Thanks for spotting this! 😅

**Review by Weetbix — APPROVED** [review]

**Review by Weetbix — APPROVED** [review]

**Pull request merged**